### PR TITLE
Fix adding dummy states in index reduction

### PR DESCRIPTION
**Pull request by perost**
- Change case in makeAllDummyVarandDummyDerivativeRepl1 to check for
  StateSelect.Prefer like the comment says.

### CODE REVIEW

**Comment by perost** [comment]
This possibly fixes #7579.

**Review by rfranke — APPROVED** [review]
The four PowerSystems examples given in #7579 work with this. Also my larger model works with the old frontend again.
The new frontend now causes the following error for the larger model:
```
Error: post-optimization module simplifyComplexFunction (initialization) failed.
```
But that's a different issue.

File: OMCompiler/Compiler/BackEnd/BackendVariable.mo [hunk 1]
```diff
@@ -515,6 +515,17 @@ algorithm
   end match;
 end varStateSelectNever;
 
+public function varStateSelectPrefer
+  "Returns true, if the state select attribute is DAE.PREFER()"
+  input BackendDAE.Var inVar;
+  output Boolean isNever;
+algorithm
+  isNever := match(varStateSelect(inVar))
```
Comment (phannebohm) [inline OMCompiler/Compiler/BackEnd/BackendVariable.mo:523]:
```suggestion 
  output Boolean isPrefer;
algorithm
  isPrefer := match(varStateSelect(inVar))
```

**Review by phannebohm — APPROVED** [review]
Looks like the correct fix to me. Congrats for finding that so quickly!

**Comment by perost** [comment]
> Looks like the correct fix to me. Congrats for finding that so quickly!

It shows the importance of comments I guess, without the comment disagreeing with the code I'd never have found the issue :)

I fixed the copy/paste issue. I guess I'll just merge it once the checks are complete, it's such a small change that we can easily revert it later if necessary.

**Pull request merged**